app/javascript/hooks/use-history.ts

```diff
@@ -11,28 +11,36 @@ function removeEmpty<TParams>(obj: TParams) {
         return v !== null
       }
     })
-    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {})
+    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {}) as TParams
```

Comment:
```suggestion
    .reduce<TParams>((acc, [k, v]) => ({ ...acc, [k]: v }), {})
```

This doesn't cast it, but only stops inferring the type. Using this will still show you errors if it's not correct (can't be narrowed), whereas the cast shows that only if it has no overlap (so no narrows and no widens).